assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/llava/utils.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Mlflow LLaVA wrapper helper functions."""
+
+import base64
+import re
+import requests
+
+from typing import Union
+
+
+def process_image(image: Union[str, bytes]) -> bytes:
+    """Process input image.
+
+    If input image is in bytes format, return it as it is.
+    If input image is in base64 string format, decode it to bytes.
+    If input image is in url format, download it and return bytes.
+    https://github.com/mlflow/mlflow/blob/master/examples/flower_classifier/image_pyfunc.py
+
+    :param image: image in base64 string format or url or bytes.
+    :type image: string or bytes
+    :return: decoded image.
+    :rtype: bytes
+    """
+    if isinstance(image, bytes):
+        return image
+    elif isinstance(image, str):
+        if _is_valid_url(image):
+            return requests.get(image).content
```

Comment:
There is a fix for better error message in case of private urls in this step, that was added to vision_utils.py.
It looks like the only difference is the input/output type. Would it be possible to reuse the common parts between the two functions.

Reply:
Same answer as before.